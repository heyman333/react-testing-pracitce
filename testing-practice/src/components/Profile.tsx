import React from "react";
import styles from "./CSSModule.module.scss";

interface Props {
  name: string;
  profileImgUrl?: string;
  age: number;
}

const CircleProfile: React.FunctionComponent<Props> = ({
  name,
  profileImgUrl,
  age,
}) => {
  return (
    <div className={styles.wrapper}>
      <div className={styles.profileWrapper} data-testid="imageWrapper">
        {profileImgUrl && (
          <img className={styles.img} src={profileImgUrl} alt="프로필이미지" />
        )}
      </div>
      <div>{name}</div>
      <div>{age}</div>
    </div>
  );
};

export default CircleProfile;
