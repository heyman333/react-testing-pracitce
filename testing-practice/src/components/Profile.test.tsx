import React from "react";
import { render } from "@testing-library/react";
import Profile from "./Profile";

describe("<Profile /> without Image interaction", () => {
  it("matches snapshot", () => {
    const utils = render(<Profile name="한영수" age={29} />);
    expect(utils.container).toMatchSnapshot();
  });

  it("shows the props correctly", () => {
    const utils = render(<Profile name="한영수" age={29} />);
    utils.getByText("한영수");
    utils.getByText("29");
  });

  it("do not have img node", () => {
    const utils = render(<Profile name="한영수" age={29} />);
    const imageContainer = utils.getByTestId("imageWrapper");
    expect(imageContainer.childElementCount).toBe(0);
  });
});

describe("<Profile /> with Image interaction", () => {
  it("matches snapshot", () => {
    const utils = render(
      <Profile
        name="한영수"
        age={29}
        profileImgUrl="https://www.fillmurray.com/360/360"
      />
    );
    expect(utils.container).toMatchSnapshot();
  });

  it("have profile image node", () => {
    const utils = render(
      <Profile
        name="한영수"
        age={29}
        profileImgUrl="https://www.fillmurray.com/360/360"
      />
    );
    utils.getByAltText("프로필이미지");
  });
});
