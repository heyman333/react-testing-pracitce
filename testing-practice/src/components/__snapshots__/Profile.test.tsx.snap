// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> with Image interaction matches snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="profileWrapper"
      data-testid="imageWrapper"
    >
      <img
        alt="프로필이미지"
        class="img"
        src="https://www.fillmurray.com/360/360"
      />
    </div>
    <div>
      한영수
    </div>
    <div>
      29
    </div>
  </div>
</div>
`;

exports[`<Profile /> without Image interaction matches snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="profileWrapper"
      data-testid="imageWrapper"
    />
    <div>
      한영수
    </div>
    <div>
      29
    </div>
  </div>
</div>
`;
